cajita/unit_test/tstParticleGridDistributor3d.hpp

```diff
@@ -262,6 +262,90 @@ void localOnlyTest( const GridType global_grid, const double cell_size )
         }
 }
 
+//---------------------------------------------------------------------------//
+// The objective of this test is to verify that particles outside the grid are
+// removed correctly if outside the global domain. In this case no particles
+// should remain in the system since they are all created outside (destination
+// rank is set to -1 if not inside the system).
+template <class GridType>
+void removeOutsideTest( const GridType global_grid )
+{
+    // Get the local block with a halo of 2.
+    const int data_halo_size = 2;
+    auto block = Cajita::createLocalGrid( global_grid, data_halo_size );
+    auto local_mesh = Cajita::createLocalMesh<Kokkos::HostSpace>( *block );
+
+    // Allocate particles
+    auto owned_cell_space =
+        block->indexSpace( Cajita::Own(), Cajita::Cell(), Cajita::Local() );
+    int num_particle = owned_cell_space.size();
+    using MemberTypes = Cabana::MemberTypes<double[3], int>;
+    using ParticleContainer = Cabana::AoSoA<MemberTypes, Kokkos::HostSpace>;
+    ParticleContainer particles( "particles", num_particle );
+    auto coords = Cabana::slice<0>( particles, "coords" );
+    auto linear_ids = Cabana::slice<1>( particles, "linear_ids" );
+
+    // Put particles in the center of every local cell.
```

Comment:
Same comment here on setup. I think this can be simplified dramatically and still achieve the same effect.